typescript/src/schema/api.ts

```diff
@@ -83,3 +91,48 @@ export class API {
     );
   }
 }
+
+function getResourceMap(
+  fd: plugin.google.protobuf.IFileDescriptorProto[]
+): ResourceMap {
+  const resourceMap: ResourceMap = {};
+  for (const fileDescriptor of fd) {
+    const messages = fileDescriptor
+      .messageType!.filter(message => message.name)
```

Comment:
A few comments related to this code:
1. Naming is a little bit confusing. Let's call the function parameter `fileDescriptors` (in plural), then we can use singular `fd` or `fileDescriptor` anywhere.
2. `messageType` can be `undefined` or `null` (if there are no messages), so let's also filter out only those fds with existing messages.

Reply:
Got it!